web-ui/src/pages/MemberProfilePage.jsx

```diff
@@ -29,6 +30,37 @@ const MemberProfilePage = () => {
   const { csrf, skills, userProfile } = state;
   const { memberId } = useParams();
   const [selectedMember, setSelectedMember] = useState(null);
+  const [supervisor, setSupervisor] = useState(null)
+  const[pdl, setPDL ] = useState(null)
+
+  useEffect(() => {
+    async function getPDLName() {
+      if (selectedMember?.pdlId) {
+        let res = await getMember(selectedMember.pdlId, csrf);
+        let pdlProfile =
+          res.payload.data && !res.error ? res.payload.data : undefined;
+        setPDL(pdlProfile ? pdlProfile.name : "");
+      }
+    }
+    if (csrf) {
+      getPDLName();
+    }
+  }, [csrf, selectedMember]);
+
+useEffect(() => {
+    async function getSupervisorName() {
+      if (selectedMember?.supervisorid) {
+        let res = await getMember(selectedMember.supervisorid, csrf);
+        let supervisorProfile =
+          res.payload.data && !res.error ? res.payload.data : undefined;
+        setSupervisor(supervisorProfile ? supervisorProfile.name : "");
+      }
+    }
+    if (csrf) {
+      getSupervisorName();
+    }
+  }, [csrf, selectedMember]);
+
```

Comment:
All of the profiles get loaded into the context on load, you can just select those instead of having to make these calls.

Reply:
Fixed. Supervisor and PDL info now selected out of state.